framework/src/interfaces/Coupleable.C

```diff
@@ -2140,10 +2140,23 @@ Coupleable::coupledIndices(const std::string & var_name) const
   return coupledVectorHelper<unsigned int>(var_name, func);
 }
 
+VariableName
+Coupleable::coupledName(const std::string & var_name, unsigned int comp) const
+{
+  if (getVar(var_name, comp))
+    return getVar(var_name, comp)->name();
+  // For constants, we'll just return the constant as the name
+  else if (MooseUtils::parsesToReal(var_name))
+    return var_name;
+  else
+    mooseError(
+        _c_name, ": Variable '", var_name, "' does not exist, yet its coupled name is requested");
+}
+
 std::vector<VariableName>
 Coupleable::coupledNames(const std::string & var_name) const
 {
-  auto func = [this, &var_name](unsigned int comp) { return getVar(var_name, comp)->name(); };
+  auto func = [this, &var_name](unsigned int comp) { return coupledName(var_name, comp); };
```

Comment:
The behavior described above would not be an issue here, as we currently do not allow mixing of variable and constant coupling in one coupled variable vector.

Reply:
this code would work even when we allow the mixing if we do not disallow coupledName on a constant